cylc/flow/flow_mgr.py

```diff
@@ -41,7 +41,7 @@
 
 def add_flow_opts(parser):
     parser.add_option(
-        "--flow", action="append", dest="flow", metavar="FLOW",
+        "--flow", action="append", dest="flow", metavar="FLOW", default=[],
```

Comment:
Rebase error?

Reply:
No.. You cannot pass `None` into a argument anymore.. 
You either include it and pass in an expected type, or just don't include the argument and it will use the default if set in the schema.

Reply:
This is command line functionality, it has nothing to do with GraphQL?!

Reply:
`--flow` is in a number of commands, no?
And the mutation includes the argument, so it can no longer be passed in as `None`/`null` ..

Reply:
This is the command line `--flow` arugment. The GraphQL "flow" argument has no interaction with this.

So there shouldn't be any need to change the default value in line with this PR right?

Reply:
We either set the default in the command option (which is read into the mutation), or we construct the mutation on-the-fly to only include option arguments/variables that are not `None` ... 
Commands are mostly GraphQL so needs to happen with this PR.

Reply:
I think this may break all the commands which take a `--flow` argument - I've documented this as a bug.

Reply:
@dwsutherland, if "flow" arguments were being provided `null` values before, then shouldn't we just change the schema to allow them (i.e. remove this line):

https://github.com/cylc/cylc-flow/blob/0d93fb1cf8efa19a680995ccee8e8409ef8fc816/cylc/flow/network/schema.py#L2079

Reply:
> Otherwise there'll probably be an inter-version compatibility issue?

Can confirm that this is handled by the code the following comment is referencing:

https://github.com/cylc/cylc-flow/pull/6478#discussion_r2086416973